most-used-words.el

```diff
@@ -50,97 +50,170 @@
 
 (require 'cl-lib)
 
+(defgroup most-used-words nil
+  "Display most used words in buffer."
+  :prefix "most-used-words-"
+  :group 'tool
+  :link '(url-link :tag "Repository" "https://github.com/udyantw/most-used-words"))
+
+(defcustom most-used-words-word-display 3
+  "Default word data to display."
+  :type 'integer
+  :group 'most-used-words)
+
+(defcustom most-used-words-display-type 'buffer
+  "The method to display most used words."
+  :type '(choice (const :tag "buffer" buffer)
+                 (const :tag "table" table))
+  :group 'most-used-words)
+
+(defvar most-used-words--buffer-name "*Most used words*"
+  "Buffer name for displaying most used words.")
+
+(defvar most-used-words--buffer nil
+  "Recorded buffer for currently doing most used words.")
+
+;;; Util
+
 (cl-defmacro most-used-words-with-view-buffer (buffer &body body)
+  "Default BUFFER for this package to display and execute BODY."
   `(progn
-     (with-current-buffer ,buffer
-       ,@body)
-     (view-buffer-other-window ,buffer
-			       nil
-			       #'kill-buffer)))
+     (with-current-buffer ,buffer ,@body)
+     (view-buffer-other-window ,buffer nil #'kill-buffer)))
+
+(defun most-used-words--form-data (word count percent)
+  "Form the plist data item with WORD, COUNT, PERCENT."
+  (list :word word :count count :percent percent))
```

Comment:
This function form a `plist` item. (Utility use)

Reply:
Okay.